okhttp/src/main/java/okhttp3/Response.kt

```diff
@@ -433,7 +433,7 @@ class Response internal constructor(
     }
 
     open fun build(): Response {
-      check(code > 0) { "code < 0: $code" }
+      check(code >= 0) { "code < 0: $code" }
```

Comment:
nit: I kinda feel this should probably throw `IllegalArgumentException`? Anyways, this is unrelated to backward compatibility. Let's discuss it in another issue/pr :)

Reply:
The function accepts no arguments though

Reply:
ah, correct..  Okay, so if `Builder#code(...)` threw an exception, the exception could be `IllegalArgumentException `, which Builder doesn't do. In our case, `Builder#build()` uses _state_ that was set via setters and throws `IllegalStateException`.

Hope my understanding is correct, thanks anyway :)